test/grouping.jl

```diff
@@ -23,8 +23,23 @@ d = DataFrame(n = 1:20, x = [3, 3, 3, 3, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2, 2, 3,
 g = groupby(d, :x, sort=true)
 @test @based_on(g, nsum = sum(:n))[:nsum] == [99, 84, 27]
 
-d = DataFrame(a = [1,1,1,2,2], b = [1,2,3,missing,missing])
+# Transform tests
+d = DataFrame(a = [1,1,1,2,2,3,3,1], 
+              b = Any[1,2,3,missing,missing,6.0,5.0,4], 
+              c = CategoricalArray([1,2,3,1,2,3,1,2]))
 g = groupby(d, :a)
-@test isequal(@transform(g, t = mean(:b))[:t], [2.0, 2.0, 2.0, missing, missing])
-@test isequal(@transform(g, t = fill(:b[1], length(:b))), [1, 1, 1, missing, missing])
+## Scalar output 
+# Type promotion Int -> Float
+@test @transform(g, t = :b[1])[1, :t] == 1.0
```

Comment:
You'd need `===` to actually check the type. But that doesn't allow checking the eltype: you need two checks, one for values and one for the type of the array. While you're at it, better also check all values in the array.

Same below.